### PR TITLE
LPS-204046 User is not directed to the current folder when changing view mode

### PR DESCRIPTION
**Pull request by rebsilva**
Related issues:

- https://liferay.atlassian.net/browse/LPS-206063
- https://liferay.atlassian.net/browse/LPS-204046

resent from https://github.com/liferay-objects/liferay-portal/pull/3048

### CODE REVIEW

**Comment by rebsilva** [comment]
this PR is blocked because i'm waiting the response of frontend infra about possible security issues

**Comment by rebsilva** [comment]
they [answered](https://liferay.slack.com/archives/CNBG06JS3/p1705588429643339?thread_ts=1705500879.774119&cid=CNBG06JS3) and there isn't a security issue :tada: but we need to see if the changes of improvements he suggested will break any tests; if that's the case, we can send as it is right now and we can change to what he suggested in another moment

**Comment by rebsilva** [comment]
i will put the tests to run, if everything is okay we can send this way and send the improvement later along with the test fixes

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:object

**Comment by rebsilva** [comment]
ci:stop

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx [hunk 1]
```diff
@@ -324,15 +324,41 @@ export default function ViewObjectDefinitions({
...
+					);
+
+					window.history.replaceState(null, '', currentUrl.href);
+
+					setSelectedObjectFolder(allObjectFolders.items[0]);
+				}
+				else {
+					const newSelectedFolder = allObjectFolders.items.find(
+						(folder) => folder.name === objectFolderNameSearchParam
+					);
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:353]: Hey @rebsilva, could you use the object prefix here?

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx [hunk 2]
```diff
@@ -324,15 +324,41 @@ export default function ViewObjectDefinitions({
...
+
+					if (newSelectedFolder) {
+						setSelectedObjectFolder(newSelectedFolder);
+					}
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:357]: Use the object prefix

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx [hunk 3]
```diff
@@ -324,15 +324,41 @@ export default function ViewObjectDefinitions({
...
-				});
+				const allObjectFolders = await API.getAllObjectFolders();
+
+				setObjectFoldersRequestInfo(allObjectFolders);
 
 				const objectDefinitions = await API.getAllObjectDefinitions();
 
 				setObjectDefinitionActions(objectDefinitions.actions);
+
+				const currentUrl = new URL(window.location.href);
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:341]: Hey @rebsilva, I can see variables using uppercase URL as `baseResourceURL` and `importObjectDefinitionURL` .
The `currentUrl` do not follow this pattern, do you think we can go on with `currentURL`

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ObjectFoldersSidebar.tsx [hunk 4]
```diff
@@ -140,6 +140,19 @@ export default function ObjectFoldersSideBar({
 						key={currentObjectFolder.name}
 						onClick={() => {
 							setSelectedObjectFolder(currentObjectFolder);
+
+							const currentUrl = new URL(window.location.href);
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ObjectFoldersSidebar.tsx:144]: If you agree with the previous change, apply the same pattern here, `currentURL`

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey @rebsilva, just a few comments 😄

**Comment by rebsilva** [comment]
ci:stop

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx (hunk 1 above)
Comment (rebsilva) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:353]: sure! i will do that

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx (hunk 2 above)
Comment (rebsilva) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:357]: ok!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx (hunk 3 above)
Comment (rebsilva) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:341]: sure, i agree with this as well

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:object

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:object

**Comment by rebsilva** [comment]
ci:report:2522692247

**Comment by rebsilva** [comment]
ci:test:sf

**Review by igor-franca — APPROVED** [review]

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:object

**Comment by rebsilva** [comment]
ci:reevaluate:1307401_8637

**Comment by rebsilva** [comment]
running tests again after testray has been fixed :cherries:

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:object

**Comment by rebsilva** [comment]
ci:test:stable

**Comment by rebsilva** [comment]
ci:test:stable

**Comment by rebsilva** [comment]
i will open another PR for this, so I'm closing this one

**Pull request closed without merging**